pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -219,6 +222,7 @@ func newQueryTripperware(
 		)
 		queryInstantMiddleware = append(
 			queryInstantMiddleware,
+			newSplitInstantQueryByIntervalMiddleware(cfg.SplitInstantQueriesByInterval > 0, cfg.SplitInstantQueriesByInterval, limits, log, engine, registerer),
```

Comment:
I don't think it should be mounted here. We enter here only if query sharding is enabled, but time splitting should be unrelated from `cfg.ShardedQueries` setting.